week9/src/Home.js

```diff
@@ -0,0 +1,121 @@
+import React,{useState, useEffect} from "react";
+import { Form } from "react-bootstrap";
+import MovieResult from "./MovieResult";
+import { ToastContainer,toast} from "react-toastify";
+import "react-toastify/dist/ReactToastify.css"
+import ReactPaginate from 'react-paginate';
+function Home() {
+    const [movies, setMovies]= useState([]);
+    const [title, setTitle]= useState('');
+    const [searchHistory]=useState([]);
+    const [page, setPage] = useState(1);
+    const [loading, setLoading] = useState(false);
+    const [hasMore, setHasMore] = useState(true);
+
+    const notFound =()=>{
+        toast.error('result not found',{
+            position: "top-right", 
+            autoClose: 5000,       
+            hideProgressBar: false, 
+            closeOnClick: true,    
+            pauseOnHover: true,    
+            draggable: false,       
+    });
+}
+const noInput =()=>{
+    toast.error('search cannot be empty',{
+        position: "top-right", 
+        autoClose: 5000,       
+        hideProgressBar: false, 
+        closeOnClick: true,    
+        pauseOnHover: true,    
+        draggable: false,       
+});
+}
+const apiError =()=>{
+    toast.error('ERROR! Check your internet connection or try again later',{
+        position: "top-right", 
+        autoClose: 5000,       
+        hideProgressBar: false, 
+        closeOnClick: true,    
+        pauseOnHover: true,    
+        draggable: false,       
+});
+}
+const searchMovie = async(e)=>{
+    e.preventDefault();
+    setLoading(true);
+    try{
+        const url = `https://api.themoviedb.org/3/search/movie?api_key=b7d68d57b175ae831b45672648c74d7b&query=${title}`        
+const res=await fetch(url);
+const data = await res.json();
+console.log(data);
+const SearchInput = /^(?=.*[a-zA-Z0-9])/;
+    if (!SearchInput.test(title)){
+      noInput();
+      return;
+    }
+setPage(page + 1);
+setMovies(data.results)
+if(data.total_results==0){
+    notFound();
+}
+setLoading(false);
+setHasMore(true);
+    }catch(e){
+   apiError();
+    }
+    
+}
+
+
+function saveUserSearch(){
+    if(title==''){
+        return;
+    }else{
+        searchHistory.push(title);
+    }
+}
+let lastFiveSearches = searchHistory.slice(-5);
+const HandleScroll = () => {
+    setPage(page + 1);
+    const url = `https://api.themoviedb.org/3/search/movie?api_key=b7d68d57b175ae831b45672648c74d7b&query=${title}&page=${page}`
+    return fetch(url)
+    .then((res)=> res.json())
+    .then((data)=>setMovies(data.results));
```

Comment:
When you get results for the next page, we currently setMovies with the results from that page. Is that what we really want to do? For infinite scroll, we want to add the newly fetched movies to the previously fetched list of movies